pkg/cfn/builder/iam_helper.go

```diff
@@ -90,28 +90,33 @@ func createRole(cfnTemplate cfnTemplate, iamConfig *api.NodeGroupIAM, managed bo
 		)
 	}
 
+	appMeshActions := []string{
+		"servicediscovery:CreateService",
+		"servicediscovery:DeleteService",
+		"servicediscovery:GetService",
+		"servicediscovery:GetInstance",
+		"servicediscovery:RegisterInstance",
+		"servicediscovery:DeregisterInstance",
+		"servicediscovery:ListInstances",
+		"servicediscovery:ListNamespaces",
+		"servicediscovery:ListServices",
+		"servicediscovery:GetInstancesHealthStatus",
+		"servicediscovery:UpdateInstanceCustomHealthStatus",
+		"servicediscovery:GetOperation",
+		"route53:GetHealthCheck",
+		"route53:CreateHealthCheck",
+		"route53:UpdateHealthCheck",
+		"route53:ChangeResourceRecordSets",
+		"route53:DeleteHealthCheck",
+	}
+
 	if api.IsEnabled(iamConfig.WithAddonPolicies.AppMesh) {
 		cfnTemplate.attachAllowPolicy("PolicyAppMesh", refIR, "*",
-			[]string{
-				"appmesh:*",
-				"servicediscovery:CreateService",
-				"servicediscovery:DeleteService",
-				"servicediscovery:GetService",
-				"servicediscovery:GetInstance",
-				"servicediscovery:RegisterInstance",
-				"servicediscovery:DeregisterInstance",
-				"servicediscovery:ListInstances",
-				"servicediscovery:ListNamespaces",
-				"servicediscovery:ListServices",
-				"servicediscovery:GetInstancesHealthStatus",
-				"servicediscovery:UpdateInstanceCustomHealthStatus",
-				"servicediscovery:GetOperation",
-				"route53:GetHealthCheck",
-				"route53:CreateHealthCheck",
-				"route53:UpdateHealthCheck",
-				"route53:ChangeResourceRecordSets",
-				"route53:DeleteHealthCheck",
-			},
+			append(appMeshActions, "appmesh:*"),
+		)
+	} else if api.IsEnabled(iamConfig.WithAddonPolicies.AppMeshPreview) {
```

Comment:
I think if both flags are allowed at the same time both actions should be included. If only one of them is going to be included in the policy the UX should fail when both flags or fields are provided (as they are incompatible).

Reply:
I've reconsidered this, both flags could be specified as one may want to switch between stable and preview on the same cluster.